drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -5679,14 +5679,15 @@ uint64_t RenderingDeviceDriverD3D12::get_resource_native_handle(DriverResource p
 			return 0;
 		}
 		case DRIVER_RESOURCE_COMMAND_QUEUE: {
-			return (uint64_t)p_driver_id.id;
+			const CommandQueueInfo *cmd_queue_info = (const CommandQueueInfo *)p_driver_id.id;
+			return (uint64_t)cmd_queue_info->d3d_queue.Get();
 		}
 		case DRIVER_RESOURCE_QUEUE_FAMILY: {
 			return 0;
 		}
 		case DRIVER_RESOURCE_TEXTURE: {
 			const TextureInfo *tex_info = (const TextureInfo *)p_driver_id.id;
-			return (uint64_t)tex_info->main_texture;
+			return (uint64_t)tex_info->resource;
```

Comment:
This might need to be `tex_info->main_texture->resource` Otherwise it is returning the handle to the texture slice, not to the texture itself

Reply:
Good point, and this matches the pattern of other parts of the code. I've fixed that now, thanks.

As an aside, it looks like in practice these are always the same resource because D3D12 doesn't have an object that maps cleanly to `VkImageView` (Metal also returns the same object for both `DRIVER_RESOURCE_TEXTURE` and `DRIVER_RESOURCE_TEXTURE_VIEW`).